authentication/utils.py

```diff
@@ -0,0 +1,79 @@
+from authentication.models import UserToken, AppUser
+import re
+from django.utils.encoding import force_bytes
+from django.utils.http import urlsafe_base64_encode
+from django.template.loader import render_to_string
+from .tokens import account_token
+from django.core.mail import EmailMessage
+import secrets, string
+
+async def send_verification_email(user, token):
+    username = user.username
+    email = user.real_email
+    subject = "Revelio - Verify Email"
+    message = render_to_string('verify_email_msg.html', {
+        'username': username,
+        'uid':urlsafe_base64_encode(force_bytes(user.pk)),
+        'token':token,
+    })
+    email = EmailMessage(
+        subject, message, to=[email]
+    )
+    email.content_subtype = 'html'
+    email.send()
+
+async def send_recover_account_email(user, token):
+    email = user.email
+    subject = "Revelio - Password Recovery Email"
+    message = render_to_string('change_password_email_msg.html', {
+        'token':token,
+    })
+    email = EmailMessage(
+        subject, message, to=[email]
+    )
+    email.content_subtype = 'html'
+    email.send()
+
+
+def validate_input(username, email, password):
+    if username is None or password is None or email is None:
+            return'One or more fields are missing!'
+    if AppUser.objects.filter(username = username).exists() or AppUser.objects.filter(email = email).exists():
+            return 'Username and/or email already taken!'
+    if username.strip() == '' or password.strip() == '' or email.strip() == '':
+        return 'Empty input! Make sure all the fields are filled.'
+    if re.match(r'^[\w\.-]+@[\w\.-]+\.\w+$', email) is None:
+        return 'Email format is wrong.'
```

Comment:
I think you can refactor this matched pattern, just in case if any future features need it.